bin/oref0-pushover.sh

```diff
@@ -28,3 +28,94 @@ elif [[ $ONLYFOR =~ "insulin" ]] && ! cat $FILE | egrep "maxBolus"; then
 else
     curl -s -F "token=$TOKEN" -F "user=$USER" -F "sound=$SOUND" -F "message=$(jq -c "{bg, tick, carbsReq, insulinReq, reason}|del(.[] | nulls)" $FILE) - $(hostname)" https://api.pushover.net/1/messages.json && touch monitor/pushover-sent
 fi
+
+# Send to pushover glances API 
+#   For watch face updates once every 10 minutes
+#   Works for apple watch complications 
+#   Not tested fo android wear notifications yet
+
+# Use file to make sure we don't update glances more often than every 10 mins
+# Apple watch disables any complication updates that happen more frequently
+
+source $HOME/.bash_profile
+key=${MAKER_KEY:-"null"}
+carbsReq=`jq .carbsReq ${FILE}`
+tick=`jq .tick ${FILE}`
+bgNow=`jq .bg ${FILE}`
+delta=`echo "${tick}" | tr -d +`
+delta="${delta%\"}"
+delta="${delta#\"}"
+cob=`jq .COB $FILE`
+iob=`jq .IOB $FILE`
+
+#echo "carbsReq=${carbsReq} tick=${tick} bgNow=${bgNow} delta=${delta} cob=${cob} iob=${iob}"
+pushoverGlances=$(cat preferences.json | jq -M '.pushoverGlances')
+
+if [ "${pushoverGlances}" == "null" -o "${pushoverGlances}" == "false" ]; then
+    echo "No preference or false preference for pushoverGlances"
+else
+  GLANCES="monitor/last_glance"
+  GLUCOSE="monitor/glucose.json"
+  if [ ! -f $GLANCES ]; then
+    # First time through it will get created older than 10 minutes so it'll fire
+    touch $GLANCES && touch -r $GLANCES -d '-11 mins' $GLANCES
+  fi
+
+  if test `find $GLANCES -mmin +10`
```

Comment:
If you have more than one rig, this will be too often.  Do we need to run it less often in that scenario to avoid burning through the complications budget for the day? Or will the watch just ignore the ones sent more than once every 10 minutes?  I wonder if this is something that we should eventually port over to NS to avoid duplicate notifications from multiple rigs.

Reply:
Good point about multiple rigs. We really only want one rig to do this work ... the one that is currently active. I would need to test it, but I think we are still safe in this scenario because the non-active rigs would have an enact time > 5 minutes right? In this case the entire script exits at the beginning.

Reply:
When running multiple rigs in the same or adjacent rooms, it's very common for one rig to enact something, and then as soon as it's done another rig reads the pump and enacts something slightly different.